DatadogTrace/Sources/OpenTelemetry/OTelSpan.swift

```diff
@@ -86,23 +86,54 @@ internal class OTelSpan: OpenTelemetryApi.Span {
         )
     }
 
-    // swiftlint:disable unavailable_function
+    /// Sends a span event which is akin to a log in Datadog
+    /// - Parameter name: name of the event
     func addEvent(name: String) {
-        fatalError("Not implemented yet")
+        addEvent(name: name, timestamp: .init())
     }
 
+    /// Sends a span event which is akin to a log in Datadog
+    /// - Parameters:
+    ///  - name: name of the event
+    /// - timestamp: timestamp of the event
     func addEvent(name: String, timestamp: Date) {
-        fatalError("Not implemented yet")
+        addEvent(name: name, attributes: .init(), timestamp: .init())
     }
 
+    /// Sends a span event which is akin to a log in Datadog
+    /// - Parameters:
+    /// - name: name of the event
+    /// - attributes: attributes of the event
+    /// - timestamp: timestamp of the event
     func addEvent(name: String, attributes: [String: OpenTelemetryApi.AttributeValue]) {
-        fatalError("Not implemented yet")
+        addEvent(name: name, attributes: attributes, timestamp: .init())
     }
 
+    /// Sends a span event which is akin to a log in Datadog
+    /// - Parameters:
+    /// - name: name of the event
+    /// - attributes: attributes of the event
+    /// - timestamp: timestamp of the event
     func addEvent(name: String, attributes: [String: OpenTelemetryApi.AttributeValue], timestamp: Date) {
-        fatalError("Not implemented yet")
+        var ended = false
+        queue.sync {
+            guard isRecording else {
+                ended = true
+                return
+            }
+        }
```

Comment:
question/ Is the plan to manage `isRecording` with lock, instead of blocking current thread with sync operation on another queue? Likely, after `tracer.queue` removal from https://github.com/DataDog/dd-sdk-ios/pull/1615 is merged this would be possible.

Reply:
Yes, once your PR is merged, I'm going to adopt those changes.